### PR TITLE
docs: decision record about renaming git branches

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## WHAT
This PR adds a decision record about why and how we should rename our Git branches.

### CODE REVIEW

**Review by ndr-brt — APPROVED** [review]

File: docs/development/decision-records/2023-04-03_renaming_branches/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+- `main` becomes our work/development branch. All pull requests target `main`.
+- `develop` gets deleted
+- a new branch `releases` is introduced, which tracks the release history and receives post-release merge commits.
+
+Technically this will involve force-pushing, which is a potentially destructive operation. Therefor the following
+section outlines the exact sequence of steps. Note that "upstream" refers to `eclipse-tractusx/tractusx-edc`, while "
+fork" refers to `catenax-ng/tx-tractusx-edc`.
+
+- create a new branch `upstream/releases`
+- create a new branch `fork/releaes`, set it to track `upstream/releases`
```
Comment (florianrusch-zf) [inline docs/development/decision-records/2023-04-03_renaming_branches/README.md:47]:
```suggestion
- create a new branch `fork/releases`, set it to track `upstream/releases`
```

File: docs/development/decision-records/2023-04-03_renaming_branches/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+- push the contents of `fork/main` -> `upstream/releases`
```
Comment (florianrusch-zf) [inline docs/development/decision-records/2023-04-03_renaming_branches/README.md:48]: Why that? We can create the `upstream/releases` directly based on `upstream/main`. Or do we have commits on `fork/main` that are not on `upstream/main`?

File: docs/development/decision-records/2023-04-03_renaming_branches/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+- synchronize `upstream/develop` with `fork/develop`
+- force-push the contents of `develop` -> `upstream/main` (do **not** update the tracking branch!)
+- synchronize `upstream/main` -> `fork/main`.
```
Comment (florianrusch-zf) [inline docs/development/decision-records/2023-04-03_renaming_branches/README.md:51]:
```suggestion
- synchronize `upstream/main` -> `fork/main`
```

**Review by florianrusch-zf — CHANGES_REQUESTED** [review]
We should also document e.g. in the CONTRIBUTING.md file, that we are following an modified git flow. But I will also add this to the issue.

**Comment by florianrusch-zf** [comment]
> We should also document e.g. in the CONTRIBUTING.md file, that we are following an modified git flow. But I will also add this to the issue.

never mind! We have documented it in the PR so that should be enough or?

File: docs/development/decision-records/2023-04-03_renaming_branches/README.md (hunk 2 above)
Comment (paullatzelsperger) [reply docs/development/decision-records/2023-04-03_renaming_branches/README.md:48]: Yup, unfortunately that's the reason

**Review by florianrusch-zf — APPROVED** [review]

**Pull request merged**